PR: https://github.com/odoo/odoo/pull/105678

From: 5cc86c3bb10756a66c7b90336e20bfc3bbe546b7
From: Anita (anko)
Date: 2023-01-20 17:54:44

Structural Changes: 6
Total Changes: 54

[IMP] payment: Rename Payment Icon to Payment Method.

Payment Icon sounds confusig comparing to what it really is,
payment method name makes it clearer for user to understand
what it is.

task-2882564

closes odoo/odoo#105678

Related: odoo/enterprise#33908
Related: odoo/upgrade#4029
Related: odoo/documentation#2955
Signed-off-by: Antoine Vandevenne (anv) <anv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/payment/models/__init__.py
+++ b/addons/payment/models/__init__.py
@@ -1,8 +1,8 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import ir_http
+from . import payment_method
 from . import payment_provider
-from . import payment_icon
 from . import payment_token
 from . import payment_transaction
 from . import res_company

--- a/addons/payment/models/payment_icon.py
+++ b/addons/payment/models/payment_method.py
@@ -3,18 +3,18 @@
 from odoo import fields, models
 
 
-class PaymentIcon(models.Model):
-    _name = 'payment.icon'
-    _description = 'Payment Icon'
+class PaymentMethod(models.Model):
+    _name = 'payment.method'
+    _description = 'Payment Method'
     _order = 'sequence, name'
 
     name = fields.Char(string="Name")
     provider_ids = fields.Many2many(
         string="Providers", comodel_name='payment.provider',
-        help="The list of providers supporting this payment icon")
+        help="The list of providers supporting this payment method")
     image = fields.Image(
         string="Image", max_width=64, max_height=64,
-        help="This field holds the image used for this payment icon, limited to 64x64 px")
+        help="This field holds the image used for this payment method, limited to 64x64 px")
     image_payment_form = fields.Image(
         string="Image displayed on the payment form", related='image', store=True, max_width=45,
         max_height=30)

--- a/addons/payment/models/payment_provider.py
+++ b/addons/payment/models/payment_provider.py
@@ -47,8 +47,8 @@ class PaymentProvider(models.Model):
         related='company_id.currency_id',
         help="The main currency of the company, used to display monetary fields.",
     )
-    payment_icon_ids = fields.Many2many(
-        string="Supported Payment Icons", comodel_name='payment.icon')
+    payment_method_ids = fields.Many2many(
+        string="Supported Payment Methods", comodel_name='payment.method')
     allow_tokenization = fields.Boolean(
         string="Allow Saving Payment Methods",
         help="This controls whether customers can save their payment methods as payment tokens.\n"
@@ -194,7 +194,7 @@ class PaymentProvider(models.Model):
     show_credentials_page = fields.Boolean(compute='_compute_view_configuration_fields')
     show_allow_tokenization = fields.Boolean(compute='_compute_view_configuration_fields')
     show_allow_express_checkout = fields.Boolean(compute='_compute_view_configuration_fields')
-    show_payment_icon_ids = fields.Boolean(compute='_compute_view_configuration_fields')
+    show_payment_method_ids = fields.Boolean(compute='_compute_view_configuration_fields')
     show_pre_msg = fields.Boolean(compute='_compute_view_configuration_fields')
     show_pending_msg = fields.Boolean(compute='_compute_view_configuration_fields')
     show_auth_msg = fields.Boolean(compute='_compute_view_configuration_fields')
@@ -246,7 +246,7 @@ class PaymentProvider(models.Model):
         - `show_credentials_page`: Whether the "Credentials" notebook page should be shown.
         - `show_allow_tokenization`: Whether the `allow_tokenization` field should be shown.
         - `show_allow_express_checkout`: Whether the `allow_express_checkout` field should be shown.
-        - `show_payment_icon_ids`: Whether the `payment_icon_ids` field should be shown.
+        - `show_payment_method_ids`: Whether the `payment_method_ids` field should be shown.
         - `show_pre_msg`: Whether the `pre_msg` field should be shown.
         - `show_pending_msg`: Whether the `pending_msg` field should be shown.
         - `show_auth_msg`: Whether the `auth_msg` field should be shown.
@@ -263,7 +263,7 @@ class PaymentProvider(models.Model):
             'show_credentials_page': True,
             'show_allow_tokenization': True,
             'show_allow_express_checkout': True,
-            'show_payment_icon_ids': True,
+            'show_payment_method_ids': True,
             'show_pre_msg': True,
             'show_pending_msg': True,
             'show_auth_msg': True,
